### PR TITLE
Gustavolbs/feat/create user register page

### PR DESCRIPTION
**Pull request by gustavolbs**
## Description

Creates user register page
- Creates Form
- Creates Page
- Adds validation
- Adds Mirage

## Related Issues

Closes [Criação de tela de cadastro de usuário](https://trello.com/c/rLWE23Wy)
Closes [Publicar frontend](https://trello.com/c/esUHVOiI)

## Progress

Use checkboxes to show that you did and what you have to do. For example:

- [x] New endpoint created;
- [ ] Update organizations;
- [ ] Create tests;

## How to test it

1. Access homepage
2. Click at "Cadastro" at navigation bar. You should be redirected.
3. You can test filling the fields with invalid data or with the Happy Path. Both should be working
4.  Register will not work for now, since we don't have API endpoint yet, but you should be available to test locally thanks to MIrage.js.

## Visual reference

Screen Shot 2022-11-10 at 11 02 51
Screen Shot 2022-11-10 at 11 03 04

## Change logs

- Updates eslint
- Adds packages
- Creates SignUpForm
- Adds Mirage.js
- Creates SignUp Page
- Creates string validation helper

### CODE REVIEW

File: src/components/SignUpForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  ) => void;
+}
+
+const SignUpForm: React.FC<ISignUpForm> = ({ setSnackbar }) => {
+  const navigate = useNavigate();
+
+  const [loading, setLoading] = useState(false);
+  const [formData, setFormData] = useState(INITIAL_DATA);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
```
Comment (viniciuslacerda0) [inline src/components/SignUpForm/index.tsx:53]: Put HandleChange and HandleChangeSelect inside an useCallback, so it won’t be recreated when page updated

File: src/components/SignUpForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  };
+
+  const handleChangeSelect = (event: SelectChangeEvent): void => {
+    setFormData(curr => ({
+      ...curr,
+      [event.target.name]: event.target.value || undefined,
+    }));
+  };
+
+  const handleSubmit = async (): Promise<void> => {
```
Comment (viniciuslacerda0) [inline src/components/SignUpForm/index.tsx:67]: Same goes here

File: src/components/SignUpForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    setLoading(true);
+    try {
```
Comment (viniciuslacerda0) [inline src/components/SignUpForm/index.tsx:69]: You can do it with then().error().finally()

File: src/components/SignUpForm/index.tsx (hunk 1 above)
Comment (gustavolbs) [reply src/components/SignUpForm/index.tsx:53]: **praise:** Nice catch!

File: src/components/SignUpForm/index.tsx (hunk 3 above)
Comment (gustavolbs) [reply src/components/SignUpForm/index.tsx:69]: No, I can't in this case. `validateString` throws errors that I need to catch before request is made

**Comment by viniciuslacerda0** [comment]
It is showing two snackbars

[Screencast 2022-11-10 15:05:07.webm](https://user-images.githubusercontent.com/29777650/201172706-ac15e7f4-0fa5-4583-ad9a-ca255dd4ddb5.webm)

**Comment by gustavolbs** [comment]
> It is showing two snackbars
> 
>  Screencast.2022-11-10.15.05.07.webm

No, it's just one snackbar, but once you click it removes the message and changes the severity, so it appears to be a second snackbar. I'll fix this.

**Comment by gustavolbs** [comment]
Fixed your comments, please, take a look again

**Review by viniciuslacerda0 — APPROVED** [review]
GJ

**Pull request merged**